n2kMapper.js

```diff
@@ -15,33 +15,55 @@ function N2kMapper (options) {
   this.state = {}
 
   this.on('n2kRequestMetadata', (src) => {
-    this.emit('n2kOut', {
-      "pgn": 59904,
-      "dst": src,
-      "PGN": 126996
-    })
-    this.emit('n2kOut', {
-      "pgn": 59904,
-      "dst": src,
-      "PGN": 126998
-    })
-    this.emit('n2kOut', {
-      "pgn": 59904,
-      "dst": src,
-      "PGN": 60928
-    })
+    if ( src === 255 ) {
+      return
+    }
+    this.requestMetaData(src, 126996)
+    this.requestMetaData(src, 126998)
+    this.requestMetaData(src, 60928)
   })
 }
 
+N2kMapper.prototype.requestMetaData = function(src, pgn) {
+  const reqPgn = {
+    "pgn": 59904,
+    "dst": src,
+    "PGN": pgn
+  }
+  let retries = 5
+  debug(`requesting pgn ${pgn} from src ${src}`)
+  this.emit('n2kOut', reqPgn)
+  let interval = setInterval(() => {
+    if ( retries-- === 0 ) {
+      debug(`did not get meta pgn ${pgn} for src ${src}`)
```

Comment:
Should this be logged as an error?

Would `metadataReceived` and `metadataTimeout` events be useful?

Reply:
I added  n2kSourceMetadataTimeout , there was already an event for received